mlos_bench/mlos_bench/services/remote/azure/azure_services.py

```diff
@@ -397,10 +372,25 @@ def _check_deployment(self, _params: dict) -> Tuple[Status, dict]:
             A pair of Status and result. The result is always {}.
             Status is one of {SUCCEEDED, PENDING, FAILED}
         """
-        _LOG.info("Check deployment: %s", self.config["vmName"])
+        config = merge_parameters(
+            dest=self.config.copy(),
+            source=params,
+            required_keys=[
+                "subscription",
+                "resourceGroup",
+                "deploymentName",
```

Comment:
nit:
Most of these required keys are repeated across `merge_parameters()` calls through the file.
Does it make sense to move those into `__init__` so that they can be caught earlier at config instantiation time, rather than at runtime and then remove them from the other sites as implicitly required and reduce some boilerplate?

Reply:
it's a bit complicated. `required_keys` are not only for validation, but also to enforce copying the data from the global config if the corresponding placeholder parameter is omitted from the config section. I'll think if there is a way to do it more elegantly and send a new PR later